Sources/Host/Darwin/LibProc.cpp

```diff
@@ -55,7 +58,23 @@ void LibProc::EnumerateProcesses(
 }
 
 std::string LibProc::GetThreadName(ProcessId pid, ThreadId tid) {
-  return "<unknown>";
+  std::string dft("<unknown-thread>");
+  thread_identifier_info_data_t threadId;
+  struct proc_threadinfo ti;
+  ErrorCode err;
+  Mach mach(pid);
```

Comment:
Yea. I did that du to bad dependency. I mean I don't since it's a huge problem, a mach object is here to describe a process, so you will need to create a new one each time you want to get specific info.